playback/src/player.rs

```diff
@@ -1321,25 +1317,30 @@ impl PlayerInternal {
                         // For the basic normalisation method, a normalisation factor of 1.0 indicates that
                         // there is nothing to normalise (all samples should pass unaltered). For the
                         // dynamic method, there may still be peaks that we want to shave off.
-                        if self.config.normalisation
-                            && !(f64::abs(normalisation_factor - 1.0) <= f64::EPSILON
-                                && self.config.normalisation_method == NormalisationMethod::Basic)
-                        {
-                            // zero-cost shorthands
-                            let threshold_db = self.config.normalisation_threshold_dbfs;
-                            let knee_db = self.config.normalisation_knee_db;
-                            let attack_cf = self.config.normalisation_attack_cf;
-                            let release_cf = self.config.normalisation_release_cf;
-
-                            for sample in data.iter_mut() {
-                                *sample *= normalisation_factor; // for both the basic and dynamic limiter
-
-                                // Feedforward limiter in the log domain
-                                // After: Giannoulis, D., Massberg, M., & Reiss, J.D. (2012). Digital Dynamic
-                                // Range Compressor Design—A Tutorial and Analysis. Journal of The Audio
-                                // Engineering Society, 60, 399-408.
-                                if self.config.normalisation_method == NormalisationMethod::Dynamic
-                                {
+                        if self.config.normalisation {
+                            if self.config.normalisation_method == NormalisationMethod::Basic
+                                && normalisation_factor < 1.0
```

Comment:
Are you sure it can't be over 1.0? What if pregain is set to a positive value?

The intention of the current code is to skip the code when normalisation would be a no-op.

Reply:
It can not be over 1.0 in basic mode.

Reply:
OK I see now that's the case even when pregain is positive. That's the check I was looking for.

Reply:
Yep. Now in basic mode unless the normalisation_factor < 1.0 it's a noop.